### PR TITLE
Add transparency option to PDF export #25407

### PR DESCRIPTION
**Pull request by LeoTaoeee**
Resolves: #25407 

- [X] I signed the [CLA](https://musescore.org/en/cla)
- [X] The title of the PR describes the problem it addresses
- [X] Each commit's message describes its purpose and effects, and references the issue it resolves
- [X] If changes are extensive, there is a sequence of easily reviewable commits
- [X] The code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [X] There are no unnecessary changes
- [X] The code compiles and runs on my machine, preferably after each commit individually
- [ ] I created a unit test or vtest to verify the changes I made (if applicable)

### CODE REVIEW

File: src/importexport/imagesexport/iimagesexportconfiguration.h
Comment (cbjeukendrup) [inline src/importexport/imagesexport/iimagesexportconfiguration.h:1]: Please make sure that there are no unnecessary whitespace changes in this file (and other files)

File: src/importexport/imagesexport/internal/imagesexportconfiguration.cpp [hunk 1]
```diff
@@ -51,6 +52,18 @@ void ImagesExportConfiguration::setExportPdfDpiResolution(int dpi)
     settings()->setSharedValue(EXPORT_PDF_DPI_RESOLUTION_KEY, Val(dpi));
 }
 
+//pdf with transparency
```
Comment (cbjeukendrup) [inline src/importexport/imagesexport/internal/imagesexportconfiguration.cpp:55]: This comment is not necessary, since the name of the method is already meaningful enough

File: src/importexport/imagesexport/internal/imagesexportconfiguration.h [hunk 2]
```diff
@@ -33,6 +33,10 @@ class ImagesExportConfiguration : public IImagesExportConfiguration
     int exportPdfDpiResolution() const override;
     void setExportPdfDpiResolution(int dpi) override;
 
+    //pdf with transparent
```
Comment (cbjeukendrup) [inline src/importexport/imagesexport/internal/imagesexportconfiguration.h:36]: (Same here)

File: src/importexport/imagesexport/internal/pdfwriter.cpp [hunk 3]
```diff
@@ -65,9 +65,15 @@ Ret PdfWriter::write(INotationPtr notation, io::IODevice& destinationDevice, con
         return false;
     }
 
+    //transparency handling
```
Comment (cbjeukendrup) [inline src/importexport/imagesexport/internal/pdfwriter.cpp:68]: This comment doesn't add much information either; we prefer only writing comments when they contain information that cannot (easily) be deduced from the code itself.

File: src/importexport/imagesexport/internal/pdfwriter.cpp [hunk 4]
```diff
@@ -65,9 +65,15 @@ Ret PdfWriter::write(INotationPtr notation, io::IODevice& destinationDevice, con
...
+    const bool TRANSPARENT_BACKGROUND = options.find(OptionKey::TRANSPARENT_BACKGROUND) != options.end()
+        ? options.at(OptionKey::TRANSPARENT_BACKGROUND).toBool()
+        : configuration()->exportPdfWithTransparentBackground();
```
Comment (cbjeukendrup) [inline src/importexport/imagesexport/internal/pdfwriter.cpp:71]: `PngWriter` uses `muse::value` for this; might be good to use that here too, for consistency.

File: src/importexport/imagesexport/internal/pdfwriter.cpp [hunk 5]
```diff
@@ -107,9 +113,15 @@ Ret PdfWriter::writeList(const INotationPtrList& notations, io::IODevice& destin
...
```
Comment (cbjeukendrup) [inline src/importexport/imagesexport/internal/pdfwriter.cpp:119]: (Same here)

File: src/project/view/exportdialogmodel.h [hunk 6]
```diff
@@ -86,7 +86,8 @@ class ExportDialogModel : public QAbstractListModel, public muse::async::Asyncab
 
     Q_PROPERTY(bool shouldDestinationFolderBeOpenedOnExport READ shouldDestinationFolderBeOpenedOnExport
                WRITE setShouldDestinationFolderBeOpenedOnExport NOTIFY shouldDestinationFolderBeOpenedOnExportChanged)
-
+    Q_PROPERTY(bool pdfTransparentBackground READ pdfTransparentBackground
+               WRITE setPdfTransparentBackground NOTIFY pdfTransparentBackgroundChanged)
```
Comment (cbjeukendrup) [inline src/project/view/exportdialogmodel.h:90]: Would be good to keep the order of the Q_PROPERTIES consistent, so this one should come after `Q_PROPERTY(int pdfResolution …`

File: src/project/qml/MuseScore/Project/internal/Export/PdfSettingsPage.qml [hunk 7]
```diff
@@ -55,4 +55,25 @@ ExportSettingsPage {
             }
         }
     }
+
+    // Add transparency option
```
Comment (cbjeukendrup) [inline src/project/qml/MuseScore/Project/internal/Export/PdfSettingsPage.qml:59]: This comment is not necessary

File: src/project/qml/MuseScore/Project/internal/Export/PdfSettingsPage.qml [hunk 8]
```diff
@@ -55,4 +55,25 @@ ExportSettingsPage {
...
+
+        onClicked: {
+            root.model.pdfTransparentBackground = !checked
+        }
+    }
+
+    StyledTextLabel {
+        width: parent.width
+        text: qsTrc("project/export", "Each page of the selected parts will be exported as a separate %1 file.").arg("PDF")
+    }
```
Comment (cbjeukendrup) [inline src/project/qml/MuseScore/Project/internal/Export/PdfSettingsPage.qml:78]: I believe this label shouldn't be here; the text is not accurate, and for PDF there are already those radio buttons to select how parts should be exported

**Review by cbjeukendrup — COMMENTED** [review]
Thanks for your PR, it looks generally good! A few minor remarks.

File: src/importexport/imagesexport/internal/pdfwriter.cpp (hunk 4 above)
Comment (LeoTaoeee) [reply src/importexport/imagesexport/internal/pdfwriter.cpp:71]: can you clarify what we should do here?

Comment (cbjeukendrup) [reply src/importexport/imagesexport/internal/pdfwriter.cpp:71]: Basically, take a look at `PngWriter` (src/importexport/imagesexport/internal/pngwriter.cpp) to see how this is done there, namely like this:
```c++
    const bool TRANSPARENT_BACKGROUND = muse::value(options, OptionKey::TRANSPARENT_BACKGROUND,
                                                    Val(configuration()->exportPngWithTransparentBackground())).toBool();
```
And then do the same thing in PdfWriter.

**Review by cbjeukendrup — APPROVED** [review]

**Comment by zacjansheski** [comment]
Tested on MacOS 14, Windows 11, Ubuntu 22.04.3. Approved
https://github.com/musescore/MuseScore/issues/25407 FIXED
